### PR TITLE
Go over Breakpoint[] documentation ...

### PR DESCRIPTION
**Pull request by rocky**
and add a docstring to disabled_breakpoint

### CODE REVIEW

**Review by mmatera — APPROVED** [review]
LGTM

File: mathics/__init__.py [hunk 1]
```diff
@@ -70,6 +70,24 @@
...
+
+       PYTHONBREAKPOINT=mathics.disabled_breakpoint mathics # other arguments
+
+    See https://docs.python.org/3/library/functions.html#breakpoint for information on
+    the Python builtin breakpoint() function
+
+    """
+    # Note that we were called. In Django and other front-ends, the
+    # print message below will appear on the console; it might not be user
+    # visable by default.
```
Comment (aravindh-krishnamoorthy) [inline mathics/__init__.py:92]:
```suggestion
    # visible by default.
```

File: mathics/__init__.py [hunk 2]
```diff
@@ -70,6 +70,24 @@
 See the documentation for the full license."""
 
 
-# Disabled breakpoint
 def disabled_breakpoint():
+    """
+    This breakpoint handler can be used as a dummy breakpoint
+    handler function which does not stop in Mathics3 when Breakpoint[] is
```
Comment (aravindh-krishnamoorthy) [inline mathics/__init__.py:76]:
```suggestion
    handler function which does not stop in Mathics3 when `Breakpoint[]` is
```

File: mathics/builtin/system.py [hunk 3]
```diff
@@ -36,99 +36,40 @@
...
-          bound. Aside from 0, 640 is the smallest value allowed.
+      This can be used for debugging the Mathics3 implementation, but \
+      if you are familiar with Python, it might assist in debugging a Mathics3 programs \
+      as well.
 
-          The initial value can be set via environment variable \
-          'DEFAULT_MAX_STR_DIGITS'. If that is not set, \
-          the default value is 7000.
+      By default, the Python debugger ('pdb') is loaded. For loading other debuggers, \
+      the change the environment variable 'PYTHONBREAKPOINT'.
```
Comment (aravindh-krishnamoorthy) [inline mathics/builtin/system.py:51]:
```suggestion
      change the environment variable 'PYTHONBREAKPOINT'.
```

File: mathics/builtin/system.py [hunk 4]
```diff
@@ -36,99 +36,40 @@
...
-    so that we can see the truncation of digits in the middle:
-    >> $MaxLengthIntStringConversion = 640
-    ## Pyston 2.3.5 returns 0 while CPython returns 640
-    ## Therefore output testing below is generic.
-     = ...
 
-    Note that setting '$MaxLengthIntStringConversion' has an effect only on Python 3.11 and later;
-    Pyston 2.x however ignores this.
+    Mathics3 also provides "breakpoint" handler 'mathics.disabled_breakpoint' which \
+    notes that a 'Breakpoint[]' was encountered, but does not stop inside Python. \
```
Comment (aravindh-krishnamoorthy) [inline mathics/builtin/system.py:56]:
```suggestion
    notes that a breakpoint was encountered, but does not stop inside Python. \
```

The breakpoint could be, e.g., reached due to a `break` debugger command as well.

File: mathics/builtin/system.py [hunk 5]
```diff
@@ -36,99 +36,40 @@
...
-    Now when we print the string value of 500! and Pyston 2.x is not used, \
-    the middle digits are removed:
-    >> 500!
-     = ...
-
-    To see this easier, manipulate the result as 'String':
-
-    >> bigFactorial = ToString[500!]; StringTake[bigFactorial, {310, 330}]
-     = ...
+    >> SetEnvironment["PYTHONBREAKPOINT" -> "mathics.disabled_breakpoint"];
```
Comment (aravindh-krishnamoorthy) [inline mathics/builtin/system.py:60]:
```suggestion
The environment variable `PYTHONBREAKPOINT` can be changed during runtime to swich the breakpoint behavior.
```

This suggestion is that one could switch the command instead to enable or `disable_breakpoint`s.

```mathematica
In[4]:= SetEnvironment["PYTHONBREAKPOINT" -> "mathics.disabled_breakpoint"];
Out[4]= None

In[5]:= Breakpoint[]
Hit disabled breakpoint.
Out[5]= Breakpoint[]

In[12]:= SetEnvironment["PYTHONBREAKPOINT" -> "trepan.api.debug"];
Out[12]= None

In[13]:= Breakpoint[]
(/home/ark/git/mathics/mathics-core/mathics/builtin/system.py:899 @32): eval
<- 899         breakpoint()
R=> None
(trepan3k)
```

**Comment by aravindh-krishnamoorthy** [comment]
Looks good to me as well...

File: mathics/builtin/system.py (hunk 4 above)
Comment (rocky) [reply mathics/builtin/system.py:56]: A debugger `break` would not trigger this. Adding to the source code `breakpoint()` would though. I will revise.

**Comment by rocky** [comment]
It looks like `TimeConstrained[]` is still flaky on MacOSX. Merging in anyway. I hope we can get this flakiness removed.

If there are further changes to the text or behavior feel free to continue this in a new PR. Thanks everyone for the reviews.

**Pull request merged**